src/codewhisperer/util/telemetryHelper.ts

```diff
@@ -273,7 +272,7 @@ export class TelemetryHelper {
         const autoTriggerType = this.sessionDecisions[0].codewhispererAutomatedTriggerType
         const language = this.sessionDecisions[0].codewhispererLanguage
         const aggregated: CodewhispererUserTriggerDecision = {
-            codewhispererSessionId: sessionId,
+            codewhispererSessionId: this.sessionDecisions[0].codewhispererSessionId,
```

Comment:
can I know when exactly is this `sessionDecisions` being reset?

Reply:
It is reset in function `resetUserTriggerDecisionTelemetry`, which is the last line of function `sendUserTriggerDecisionTelemetry`

Reply:
ok, but if the passed `sessionId` is empty, this means the caller `recordUserDecisionTelemetry` will also have this empty sessionId and send it with the `userDecision` event/

Reply:
From my previous investigation, the pre-existing UserDecision telemetry has less than 1% empty `sessionId`. 

This change is not perfect, however, while being a small change, it can effectively reduce the percentage of empty session id in Aggregated User Decision events. According to my manual testing, without this change, if you press Esc at high frequency after invoking, there is a great chance of seeing one with empty session id. With this change, I have yet found a case when session id is not empty. 

Removing those global states will take much more work.